ENH: Allow refitting HPI and dev_head_t

larsoner opened the pull request:
Sometimes the Neuromag system screws up the initial HPI fitting, which then messes up `dev_head_t` and also any subsequent continuous head position estimation. This PR will fix that. Lots of TODOs in the code but I wanted to get it up to have a place to track progress.

- [x] Get it working with problematic dataset
- [x] Add test to show necessity of `if proj_op is None` addition
- [x] Show that it's essentially a no-op on correct data
- [x] Add `whats_new` entry
- [x] Add elements from problematic dataset to tests
- [x] Show it fails without refitting
- [x] Show it works with refitting
- [x] Ensure all TODOs inline are addressed

drammock commented on mne/chpi.py:
```diff
@@ -1615,3 +1625,310 @@ def get_active_chpi(raw, *, on_missing="raise", verbose=None):
+        raise ValueError(
+            "If locs is True, order must also be True (otherwise "
+            "recomputing locations has no effect)"
+        )
```
I'm +0.5 for adding this info to the docstring as well, just to save diligent readers one cycle of run-error-debug-rerun

drammock reviewed (approved):
just a few comments/suggestions/questions.  Didn't look too closely at the tests.

larsoner commented:
Thanks @drammock I think I addressed the comments! I'll mark for merge-when-green to sneak it in for 1.11. I think it should be good enough for people to try
